src/components/AddressSearch.js

```diff
@@ -107,14 +116,20 @@ const AddressSearch = (props) => {
                     if (!_.isEmpty(googlePlacesRef.current.getAddressText()) && !isTextValid) {
                         saveLocationDetails({});
                     }
+
+                    // If the text is empty, we set displayListViewBorder to false to prevent UI flickering
+                    if (_.isEmpty(text)) {
+                        setDisplayListViewBorder(false);
+                    }
                 },
             }}
             styles={{
                 textInputContainer: [styles.flexColumn],
                 listView: [
-                    styles.borderTopRounded,
-                    styles.borderBottomRounded,
-                    styles.mt1,
+                    !displayListViewBorder && styles.googleListView,
```

Comment:
By doing this, we are applying `transform: {[scale:0]}` whenever `displayListViewBorder` is false correct? Is this what you were referring to here? 

> I put up a workaround to hide the element for those few milliseconds - it's not a pretty solution 😞

Reply:
Yes! I used this workaround to prevent an UI flicker.

Reply:
Ok, gotcha. I know this has been a debugging marathon, but I think we should try and get to the bottom of why the top border flicker is happening. 

> I would imagine that we will redo a few of these inputs as part of creating our new form component, and I would suggest that we create our own version of the AddressSearch component.

I think the reality of this situation is that we are not guaranteed that we'll ever make our own `AddressSearch` component. And if we can fix it here and make it work perfectly for us, we'll never have to. I know it feels like we're wrestling with jankyness we didn't create (which is true) but we're _so close_ to finally polishing off this component. 

I'd be happy to help with debugging this. I won't have time to do that tonight, but I can check out this branch and poke around tomorrow if you'd like another pair of eyes on this.

Reply:
Fair enough 👍  My best guess for why the flicker happens is:

1. User starts typing and we fetch results. The height of the dropdown changes and it becomes visible.
2. We detect the height change, set `displayListViewBorder=true` and apply the styles.

So I think that the component becomes visible before we can apply any style changes. This might be a limitation of my solution. 

I explored other ways of preventing the flicker, but couldn't come up with anything. Feel free to take a look 😄

Reply:
What is the reason we are not considering a fork and just add an extra condition in this `if` statement:

https://github.com/FaridSafi/react-native-google-places-autocomplete/blob/47d7223dd48f85da97e80a0729a985bbbcee353f/GooglePlacesAutocomplete.js#L751-L757

We would just have to add something like: `dataSource.length &&`. It work right away and would save a bunch of time spent on workarounds.

Reply:
I'm not sure I have the answer to that question. Maybe @Luke9389 has more context here?

Reply:
So far, we haven't been faced with a situation that requires a fork, but I'd support that if it were proposed. Would either of you like to spearhead that? I'm thinking it would probably come in the form of a Problem Solution statement.